include/perf_watcher.h

```diff
@@ -12,13 +12,15 @@
 struct PerfWatcherOptions {
   bool is_kernel;
   bool is_freq;
+  uint8_t nb_frames_to_skip; // number of frames to skip during unwinding
```

Comment:
Nice, So the number of frames to skip depends on the library mechanism we implement for a given type of event. This means we should not have a different number of frames between malloc and realloc for instance.
(not saying anything, just trying to make sure I understand)

Reply:
I am in the process of changing this since it might depend on optimization level. The idea will be to skip all the frames belonging to same mapping.

Reply:
That is a good idea. Though if someone compiles with a static library, this no longer works ? I think the current logic is maybe OK.

Reply:
Indeed, this won't work with static library.
No even sure than allocation profiling will work with static lib.
Need to think a bit more on this !

Reply:
Tested allocation profiling with static lib and it worked (did no tried by a statically linked jemmalloc/tcmalloc).
I will bring back a fixed number of frames to skip that depends on compilation mode (release/LTO)...

Reply:
The idea is to skip frames particular to the profiling library, right?  I have a sense that indicating a static number of frames to skip is difficult, but I wonder whether we can emit a special frame (or detect our own entrypoints, which we could mark somehow) and we would use it to drop all stored unwinding state and basically "start over" until after that frame.  Is that idea too naive?

Reply:
Indeed, that is a great idea. We could have profiling conventions for frames. Though I'm not too worried about this. I'm sure we can fix this later.

Reply:
I was reluctant to base the logic on the name of the frames (what if symbols are stripped?).

Reply:
Relying on the entry points would work (malloc/realloc/calloc/...) but is a bit of a pain (there a 10 different entry points).